app/containers/ItunesProvider/TrackDetailContainer/tests/index.test.js

```diff
@@ -6,59 +6,102 @@
     expect(getByTestId('audio-src').src).toBe(data.previewUrl);
   });
-  it('should render track price', () => {
-    const price = 'Price : $' + data.trackPrice;
+
+  it('should render pause btn', () => {
     const { getByTestId } = renderProvider(
-      <BrowserRouter>
-        <TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />
-      </BrowserRouter>
+      <TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy}></TrackDetailContainer>
     );
-    expect(getByTestId('price').textContent).toBe(price);
+    const ele = getByTestId('pause-btn');
+    expect(ele).toBeTruthy();
   });
-  it('should render track genre', () => {
-    const genre = 'Genre : ' + data.primaryGenreName;
-    const { getByTestId } = renderProvider(
-      <BrowserRouter>
-        <TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />
-      </BrowserRouter>
+
+  it('should show - track not found when track data is empty', () => {
+    const { getByTestId } = renderProvider(<TrackDetailContainer dispatchTrackDetail={submitSpy} />);
+    expect(getByTestId('not-found')).toBeTruthy;
+  });
+
+  it('should call dispatchTrackDetail on mount', async () => {
+    renderProvider(<TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />);
+    await timeout(500);
+    expect(submitSpy).toBeCalled();
+  });
+
+  it('should dispatchTrackDetail match to the props', () => {
+    const dispatch = jest.fn();
+    mapDispatchToProps(dispatch).dispatchTrackDetail({ trackDetail: 1440649635 });
+    expect(dispatch).toHaveBeenCalled();
+  });
+
+  it('should invoke dispatchTrackDetail when params.trackId is present', async () => {
+    const submitSpy = jest.fn();
+    jest.spyOn(require('react-router-dom'), 'useParams').mockReturnValue({ trackId: null });
+    renderProvider(<TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />);
+    await timeout(500);
+    expect(submitSpy).toHaveBeenCalledTimes(0);
+  });
+  it('should invoke dispatchTrackDetail when params.trackId is present', async () => {
+    const submitSpy = jest.fn();
+    jest.spyOn(require('react-router-dom'), 'useParams').mockReturnValue({ trackId: 10 });
+    const { TrackDetailContainerTest: TrackDetailContainer } = require('../index');
+    renderProvider(<TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />);
+    await timeout(500);
+    expect(submitSpy).toHaveBeenCalledTimes(1);
+  });
+  it('should call play the track on button click', () => {
+    const playBtnSpy = jest.spyOn(require('react'), 'useRef').mockReturnValue({ current: { play: jest.fn() } });
+    const { getByTestId } = renderProvider(<TrackDetailContainer trackData={data} dispatchTrackDetail={submitSpy} />);
+    const playBtn = getByTestId('play-btn');
+    fireEvent.click(playBtn);
+    expect(playBtnSpy).toHaveBeenCalled();
```

Comment:
not sure what this test is intending to test. 
```
const playBtn = getByTestId('play-btn');
fireEvent.click(playBtn);
```
this will always make the test pass. Maybe you're trying to test some side effect?